src/ArtemisNetClient/Builders/ConnectionBuilder.cs

```diff
@@ -81,5 +102,62 @@ private void OnClosed(IAmqpObject sender, Error error)
                 _tcs.TrySetException(new CreateConnectionException(error.Description, error.Condition));
             }
         }
+        
+        private bool InternalValidateRemoteCertificate(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
+        {
+            _logger.LogInformation("Received remote certificate. Subject: {Subject} {Issuer}, Policy errors: {SslPolicyErrors}",
+                certificate.Subject,
+                certificate.Issuer,
+                sslPolicyErrors);
+
+            if (_bypassRemoteCertificateValidation)
+            {
+                _logger.LogWarning("Bypass validation for remote certificate {Subject} {Issuer}", certificate.Subject, certificate.Issuer);
+                return true;
+            }
+
+            if (sslPolicyErrors == SslPolicyErrors.None)
+            {
+                return true;
+            }
+
+            if (sslPolicyErrors == SslPolicyErrors.RemoteCertificateChainErrors)
+            {
+                _logger.LogInformation("Validate Remote Certificate against trusted remote CAs");
+
+                // check certificate in chain against trusted ca
+                if (_trustedRemoteCertificateAuthorities == null)
+                    return false;
+
+                foreach (var remote in _trustedRemoteCertificateAuthorities)
+                {
+                    var ca = new X509Certificate2(remote);
+                    var hasMatch = chain.ChainElements
+                        .Cast<X509ChainElement>()
+                        .Any(x => x.Certificate.Thumbprint == ca.Thumbprint);
+
+                    if (hasMatch)
+                    {
+                        _logger.LogInformation("Found valid CA {CertificateAuthority} for {Certificate} {Issuer}", ca.Subject, certificate.Subject, certificate.Issuer);
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private X509Certificate SelectFirstLocalCertificate(object sender, string targetHost, X509CertificateCollection localCertificates, X509Certificate remoteCertificate, string[] acceptableIssuers)
```

Comment:
I'd take a similar approach as for `InternalValidateRemoteCertificate`. We should provide a default implementation but the user should be able to override this callback.